core/src/test/scala/org/apache/spark/ExecutorAllocationManagerSuite.scala

```diff
@@ -272,17 +272,41 @@ class ExecutorAllocationManagerSuite
   test("ignore task end events from completed stages") {
     sc = createSparkContext(0, 10, 0)
     val manager = sc.executorAllocationManager.get
+
+    // We simulate having a stage fail, but with tasks still running.  Then another attempt for
+    // that stage is started, and we get task completions from the first stage attempt.  Make sure
+    // the value of `totalTasksRunning` is consistent as tasks finish from both attempts (we only
+    // count tasks from the active attempt)
```

Comment:
the last part of this comment needs to be updated -- I guess it should now be "we count all running tasks, from the zombie & non-zombie attempts"